### PR TITLE
Add git support for playbookgenerator

### PR DESCRIPTION
**Pull request by dprince**

### CODE REVIEW

File: controllers/openstackplaybookgenerator_controller.go [hunk 1]
```diff
@@ -287,6 +287,11 @@ func (r *OpenStackPlaybookGeneratorReconciler) Reconcile(ctx context.Context, re
 		requeue, err := common.WaitOnJob(job, r.Client, r.Log)
 		r.Log.Info("Generating Playbooks...")
 		if err != nil {
+			// the job failed in error
+			deleteErr := r.Client.Delete(context.TODO(), heat)
+			if err != nil {
```
Comment (abays) [inline controllers/openstackplaybookgenerator_controller.go:292]: Should this be `deleteErr` instead of `err`?

Comment (dprince) [reply controllers/openstackplaybookgenerator_controller.go]: Yes. Just pushed an update to fix this.

File: templates/openstackplaybookgenerator/bin/create-playbooks.sh [hunk 2]
```diff
@@ -132,5 +147,19 @@ ansible.write_default_ansible_cfg(
 
 EOF_PYTHON
 
-# copy to our persistent volume mount so it shows up in the 'openstackclient' pod
-cp -a /home/cloud-admin/ansible/* /var/cloud-admin/ansible/
+TMP_DIR=$(mktemp -d)
+git clone $GIT_URL $TMP_DIR
+pushd $TMP_DIR
```
Comment (bogdando) [inline templates/openstackplaybookgenerator/bin/create-playbooks.sh:152]: without popd, wouldn't this end up with the dirs stack growing up trashed with tempdirs?

File: api/v1beta1/openstackclient_types.go [hunk 3]
```diff
@@ -27,6 +27,8 @@ type OpenStackClientSpec struct {
 	ImageURL string `json:"imageURL"`
 	// name of secret holding the stack-admin ssh keys
 	DeploymentSSHSecret string `json:"deploymentSSHSecret"`
+	// GitSecret used to pull playbooks into the openstackclient pod
```
Comment (bogdando) [inline api/v1beta1/openstackclient_types.go:30]: nit: create-playbooks.sh pushes by git_url and git secret as well

File: config/manager/kustomization.yaml [hunk 4]
```diff
@@ -5,4 +5,4 @@ kind: Kustomization
 images:
 - name: controller
   newName: quay.io/openstack-k8s-operators/osp-director-operator
-  newTag: 0.0.1
+￼ newTag: 0.0.1
```
Comment (stuggi) [inline config/manager/kustomization.yaml:8]: is this to change a bad char, or introduce it :)

File: templates/openstackplaybookgenerator/bin/create-playbooks.sh (hunk 2 above)
Comment (dprince) [reply templates/openstackplaybookgenerator/bin/create-playbooks.sh:152]: The entire pod is ephemeral and is destroyed (we are using emptydir for local storage with these pods). So I think its okay

File: config/manager/kustomization.yaml (hunk 4 above)
Comment (dprince) [reply config/manager/kustomization.yaml]: yes, it is a bad char. I think its gone in the latest commit.

**Comment by abays** [comment]
This seems to be ready for merge, or did you have other changes you wanted to add @dprince?

**Comment by dprince** [comment]
I just pushed a few README and sample config changes but nothing major in that last commit. Functionally what is here works great with the dev-tools patch.

There is one reconciliation bug I am tracking down. The Ephemeral Heat is getting created and immediately deleted (which looks bad if you notice it but doesn't cause a functional issue). Other than that this can land and I think is in good shape

**Review by stuggi — APPROVED** [review]
/lgtm

**Pull request merged**